packages/devtools_app/test/vm_developer/object_inspector/object_viewport_test.dart

```diff
@@ -144,16 +50,98 @@ void main() {
     expect(find.byType(HistoryViewport<VmObject>), findsOneWidget);
   });
 
-  testWidgetsWithWindowSize('test for class Object', windowSize,
-      (WidgetTester tester) async {
-    testObjectInspectorViewController.fakeObjectHistory
-        .setCurrentObject(mockClassObject);
-    await tester.pumpWidget(
-      wrap(ObjectViewport(controller: testObjectInspectorViewController)),
-    );
-    expect(viewportTitle(mockClassObject), 'Class FooClass');
-    expect(find.text('Class FooClass'), findsOneWidget);
-    expect(find.byType(VmClassDisplay), findsOneWidget);
+  group('test for class object:', () {
+    late MockClassObject mockClassObject;
+
+    setUp(() {
+      mockClassObject = MockClassObject();
+
+      mockVmObject(mockClassObject);
+      when(mockClassObject.name).thenReturn(testClass.name);
+      when(mockClassObject.ref).thenReturn(testClass);
+      when(mockClassObject.obj).thenReturn(testClass);
+      when(mockClassObject.instances).thenReturn(null);
+    });
+
+    testWidgets('viewport shows class display', (WidgetTester tester) async {
+      testObjectInspectorViewController.fakeObjectHistory
+          .setCurrentObject(mockClassObject);
+      await tester.pumpWidget(
+        wrap(ObjectViewport(controller: testObjectInspectorViewController)),
+      );
+      expect(viewportTitle(mockClassObject), 'Class FooClass');
+      expect(find.text('Class FooClass'), findsOneWidget);
+      expect(find.byType(VmClassDisplay), findsOneWidget);
+    });
+  });
+
+  group('test for field object:', () {
+    late MockFieldObject mockFieldObject;
+
+    setUp(() {
+      mockFieldObject = MockFieldObject();
+
+      mockVmObject(mockFieldObject);
+      when(mockFieldObject.name).thenReturn(testField.name);
+      when(mockFieldObject.ref).thenReturn(testField);
+      when(mockFieldObject.obj).thenReturn(testField);
+      when(mockFieldObject.guardClass).thenReturn(null);
+      when(mockFieldObject.guardNullable).thenReturn(null);
+      when(mockFieldObject.guardClassKind).thenReturn(null);
```

Comment:
Can we make these part of `mockVmObject` if they are the defaults we want to set?

Reply:
These are specific to the FieldObject

Reply:
Done